libs/designer-ui/src/lib/html/plugins/toolbar/helper/__test__/util.spec.ts

```diff
@@ -13,20 +13,63 @@ import type { ValueSegment } from '@microsoft/designer-client-services-logic-app
   describe('cleanHtmlString', () => {
-    it.each([
-      ['<p>text1<span>\n</span>text2</p>', '<p>text1<br>text2</p>'],
-      ['<p>text</p>', '<p>text</p>'],
-      ['<p>text1</p><p><br></p><p>text2</p>', '<p>text1</p><br><p>text2</p>'],
-      ['<p>text1<br></p><p><br></p><p>text2</p>', '<p>text1</p><br><br><p>text2</p>'],
-      ['<span>text</span>', 'text'],
-      ['<span id="$[abc,variables(\'abc\'),#770bd6]$">text</span>', '<span id="$[abc,variables(\'abc\'),#770bd6]$">text</span>'],
-      ['<span id="$[concat(),concat(\'&lt;\'),#ad008c]$">text</span>', '<span id="$[concat(),concat(\'&lt;\'),#ad008c]$">text</span>'],
-      ['<span id="$[concat(),concat(\'%26lt;\'),#ad008c]$">text</span>', '<span id="$[concat(),concat(\'%26lt;\'),#ad008c]$">text</span>'],
-      ['<span id="@{variables(\'abc\')}">text</span>', '<span id="@{variables(\'abc\')}">text</span>'],
-      ['<span id="@{concat(\'&lt;\')}">text</span>', '<span id="@{concat(\'&lt;\')}">text</span>'],
-      ['<span id="@{concat(\'%26lt;\')}">text</span>', '<span id="@{concat(\'%26lt;\')}">text</span>'],
-    ])('should properly convert HTML: %p', (input, expected) => {
-      expect(cleanHtmlString(input)).toBe(expected);
+    describe('when isValuePlaintext is true', () => {
+      it.each([
+        ['<p><br></p>', '<br>'],
+        ['Hello\nWorld', 'Hello\nWorld'],
+        ['<span>Hello</span> World', 'Hello World'],
+        ['<p><br><br></p>', '<br><br>'],
+        ['<br></p>', '</p><br>'],
+        ['<p>Hello<br>World</p>', '<p>Hello<br>World</p>'],
+        ['<h1><br></h1>', '<br>'],
+        ['<h2><br><br></h2>', '<br><br>'],
+        ['<h3><br><br><br></h3>', '<br><br><br>'],
+        ['<h4><br><br><br><br></h4>', '<br><br><br><br>'],
+        ['<p>text1<br>text2</p>', '<p>text1<br>text2</p>'],
+        ['<p>text</p>', '<p>text</p>'],
+        ['<p>text1</p><p><br></p><p>text2</p>', '<p>text1</p><br><p>text2</p>'],
+        ['<p>text1<br></p><p><br></p><p>text2</p>', '<p>text1</p><br><br><p>text2</p>'],
+        ['<span>text</span>', 'text'],
+        ['<span id="$[abc,variables(\'abc\'),#770bd6]$">text</span>', '<span id="$[abc,variables(\'abc\'),#770bd6]$">text</span>'],
+        ['<span id="$[concat(),concat(\'&lt;\'),#ad008c]$">text</span>', '<span id="$[concat(),concat(\'&lt;\'),#ad008c]$">text</span>'],
+        [
+          '<span id="$[concat(),concat(\'%26lt;\'),#ad008c]$">text</span>',
+          '<span id="$[concat(),concat(\'%26lt;\'),#ad008c]$">text</span>',
+        ],
+        ['<span id="@{variables(\'abc\')}">text</span>', '<span id="@{variables(\'abc\')}">text</span>'],
+        ['<span id="@{concat(\'&lt;\')}">text</span>', '<span id="@{concat(\'&lt;\')}">text</span>'],
+        ['<span id="@{concat(\'%26lt;\')}">text</span>', '<span id="@{concat(\'%26lt;\')}">text</span>'],
+      ])('should properly convert HTML: %p', (input, expected) => {
+        expect(cleanHtmlString(input, /* isValuePlaintext */ true)).toBe(expected);
```

Comment:
One way you can avoid having to specify `/* isValuePlaintext */` is by doing the "options object" pattern, if you like.

![image](https://github.com/Azure/LogicAppsUX/assets/1350074/bd9e60d5-f747-430e-a78f-7465d9e0b9a9)

Reply:
`libs\designer-ui\src\lib\editor\base\utils\parsesegments.ts` also has examples of this.

Reply:
Will leave as is, should be clear what is being tested.

Reply:
I don't think the point here is that it's unclear what's being tested. It's unclear what true/false here means when it's being used in live code. Using booleans as a parameter is generally bad form and leads to bad readability/maintainability. That's not to say we don't have it in other parts of code(generally code that was carried over from old designer. But it would be better if it was clearer and an options object is quite good for that.